### PR TITLE
aws-ec2: se max_ttl when ttl is not set, during login

### LINKED ISSUE #1748: max_ttl ignored on aws_ec2 roles?
This is tested with 6.1rc3. I use aws_ec2 authentiation and have a role with a max_ttl of 1h:
```
# vault read /auth/aws-ec2/role/role_name
Key                      	Value
---                      	-----
allow_instance_migration 	false
bound_account_id         	REACTED
bound_ami_id             	
bound_iam_role_arn       	
disallow_reauthentication	false
max_ttl                  	3600
policies                 	[default xxx]
role_tag                 	vault_role
ttl                      	0
```

But when a machine authenticates with this role it gets a token like this:
```
# vault token-lookup
Key             	Value
---             	-----
accessor        	REDACTED
creation_time   	1471506447
creation_ttl    	2592000
display_name    	aws-ec2
explicit_max_ttl	0
id              	REDACTED
meta            	map[role_tag_max_ttl:0s ami_id:REDACTED instance_id:REDACTED region:us-east-1 role:role_name]
num_uses        	0
orphan          	true
path            	auth/aws-ec2/login
policies        	[default xxx]
renewable       	true
ttl             	2591996
```

I had expected the ttl to be 3600 since that is what the role specifies. The role tag does not specify any ttl.

Is this a bug or am I confused?

### PR DESCRIPTION
**Pull request by vishalnayak**
Fixes #1748

### CODE REVIEW

**Comment by LLBennett** [comment]
LGTM.

File: builtin/credential/aws-ec2/path_login.go [hunk 1]
```diff
@@ -406,10 +406,15 @@ func (b *backend) pathLoginUpdate(
...
 	// Cap the TTL value.
-	if shortestMaxTTL < roleEntry.TTL {
-		resp.AddWarning(fmt.Sprintf("Role ttl of %d exceeded the effective max_ttl of %d; ttl value is capped appropriately", roleEntry.TTL/time.Second, shortestMaxTTL/time.Second))
-		resp.Auth.TTL = shortestMaxTTL
+	shortestTTL := b.System().DefaultLeaseTTL()
+	if roleEntry.TTL > time.Duration(0) && roleEntry.TTL < shortestTTL {
+		shortestTTL = roleEntry.TTL
 	}
+	if shortestMaxTTL < shortestTTL {
+		resp.AddWarning(fmt.Sprintf("Effective ttl of %d exceeded the effective max_ttl of %d; ttl value is capped appropriately", shortestTTL/time.Second, shortestMaxTTL/time.Second))
```
Comment (jefferai) [inline builtin/credential/aws-ec2/path_login.go]: Maybe use %s and .String() here to be pretty printed?

Comment (vishalnayak) [inline builtin/credential/aws-ec2/path_login.go]: Sure

**Comment by jefferai** [comment]
One comment, other than that, LGTM!

**Pull request merged**